google/cloud/forseti/scanner/scanners/audit_logging_scanner.py

```diff
@@ -0,0 +1,161 @@
+                'violation_type': violation.violation_type,
+                'violation_data': violation_data,
+                'resource_data': violation.resource_data
+            }
+
+    def _output_results(self, all_violations):
+        """Output results.
+
+        Args:
+            all_violations (list): A list of violations
+        """
+        all_violations = list(self._flatten_violations(all_violations))
+        self._output_results_to_db(all_violations)
+
+    def _find_violations(self, audit_logging_data):
+        """Find violations in the audit log configs.
+
+        Args:
+            audit_logging_data (list): audit log data to find violations in.
+
+        Returns:
+            list: A list of all violations
+        """
+        all_violations = []
+        LOGGER.info('Finding audit logging violations...')
+
+        for project, audit_config in audit_logging_data:
+            violations = self.rules_engine.find_violations(
+                project, audit_config)
+            LOGGER.debug(violations)
+            all_violations.extend(violations)
+        return all_violations
+
+    def _retrieve(self):
+        """Retrieves the data for scanner.
+
+        Returns:
+            list: List of projects' audit logging config data.
+        """
+        model_manager = self.service_config.model_manager
+        scoped_session, data_access = model_manager.get(self.model_name)
+        with scoped_session as session:
+            project_configs = []
+            ancestor_configs = {}
+            # Types that can contain AuditConfigs in the IamPolicy.
+            audit_policy_types = frozenset([
+                'organization', 'folder', 'project'])
+
+            for policy in data_access.scanner_iter(session, 'iam_policy'):
+                if policy.parent.type not in audit_policy_types:
+                    continue
+                audit_config = iam_policy.IamAuditConfig.create_from(
+                    json.loads(policy.data).get('auditConfigs', []))
+
+                if policy.parent.type == 'project':
+                    project_configs.append(
+                        (Project(policy.parent.name,
+                                 policy.parent.full_name,
+                                 policy.data),
+                         audit_config))
+                elif audit_config.service_configs:
+                    ancestor_configs[policy.parent.type, policy.parent.name] = (
+                        audit_config)
+
+        # Merge ancestor configs into project level configs.
+        for project, audit_config in project_configs:
+            ancestors = utils.get_resources_from_full_name(project.full_name)
+            next(ancestors)
+            for res_type, res_name in ancestors:
+                ancestor_config = ancestor_configs.get((res_type, res_name))
+                if ancestor_config:
+                    audit_config.merge_configs(ancestor_config)
+
+        return project_configs
```

Comment:
We could add fields like "applies_to" and "inherit_from_parents" in iam_rules. I left them out in this iteration since I wasn't sure of the demand for it.